tabpfn_client/client.py

```diff
@@ -32,6 +36,80 @@ def filter(self, record):
         return True
 
 
+class DatasetCacheManager:
```

Comment:
Thanks for the review, good catch with the bug! I now moved the last train set part to the estimators, I feel like this is more straightforward with our current implementation than including it in the InferenceClient. I also did some renaming to avoid confusion. :)